files/class-api-client.php

```diff
@@ -152,14 +153,22 @@ private function calculate_upload_timeout( $file_size ) {
 
 	public function get_file( $file_path ) {
 		// check in cache first
-		$file_content = $this->cache->get_file( $file_path );
-		if ( $file_content ) {
-			return $file_content;
+		$file = $this->cache->get_file( $file_path );
+		if ( $file ) {
+			return $file;
 		}
 
+		$tmp_file = tempnam( get_temp_dir(), 'vip' );
+		$tmp_handle = fopen( $tmp_file, 'w+' );
+
+		$curl_streamer = new Curl_Download_Streamer( $tmp_handle );
```

Comment:
I just noticed that `WP_Http` has a `stream` option that already does this so the `Curl_Download_Streamer` is kind of unnecessary:

```
'stream'
(bool) Whether to stream to a file. If set to true and no filename was given, it will be droped it in the WP temp dir and its name will be set using the basename of the URL. Default false.
'filename'
(string) Filename of the file to write to when streaming. $stream must be set to true. Default null.
```

https://developer.wordpress.org/reference/classes/wp_http/request/

Reply:
It works! The streamer class is now no longer necessary! Changed in a448e535a538d6752d38c434bca0af0001a238b8

Reply:
Awesome!